go-controller/pkg/ovn/namespace.go

```diff
@@ -151,6 +153,18 @@ func (oc *Controller) AddNamespace(ns *kapi.Namespace) {
 		}
 	}
 
+	hybridOverlayExternalGW := net.ParseIP(ns.GetAnnotations()[hotypes.HybridOverlayExternalGw])
+	if hybridOverlayExternalGW == nil {
+		klog.Errorf("Could not parse hybrid overlay external gw annotation")
```

Comment:
you only want to log an error if it's _set_ but unparseable (as opposed to `""`).
and the temporary `hybridOverlayExternalGW` variable here isn't needed; you can just set `nsInfo.hybridOverlayExternalGW` directly. (Though you might want a temporary variable for the annotation value. Oh, you don't need to use `GetAnnotations()`, you should be able to use `Annotations` directly.)

Reply:
yeah, java-isms you know :-)